src/main/java/com/google/devtools/build/lib/starlarkbuildapi/StarlarkActionFactoryApi.java

```diff
@@ -67,6 +67,22 @@ public interface StarlarkActionFactoryApi extends StarlarkValue {
           + " volatile nature the output file of this action is special cased in Bazel, similar to"
           + " <code>ctx.version_file</code>. Therefore changes in this file will not retrigger"
           + " actions depending on it.";
+  static final String TOOLS_ARG_DOC = """
+List or <a href="../builtins/depset.html"><code>depset</code></a> of any tools needed by the \
+action. Tools are executable inputs that may have their own runfiles which are automatically made \
+available to the action. \
+<p>
+When a list is provided, it can be a heterogenous collection of:
+<ul>
+    <li><code>File</code>s</li>
+    <li><code>FilesToRunProvider</code> instances</li>
+    <li><code>depset</code>s of <code>File</code>s</li>
+</ul>
+Executable <code>File</code>s (created by another target) and <code>FilesToRunProvider</code>s \
```

Comment:
This lacks the nuance that `File`s only have their runfiles added if they come from a `ctx.executable` arg.